inspire_matcher/core.py

```diff
@@ -241,3 +235,96 @@ def _compile_nested_prefix(query, record):
 
 def _get_common_path(paths):
     return '.'.join(os.path.commonprefix([path.split('.') for path in paths]))
+
+
+def _compile_authors_query(query, record):
+    parsed_name = ParsedName(record.get("full_name"))
+    nested_query = _return_empty_nested_query("authors")
+    if "inner_hits" in query:
+        nested_query['query']['nested']['inner_hits'] = query['inner_hits']
+    if len(parsed_name) == 1 and '.' not in parsed_name.first:
+        nested_query['query']['nested']['query']['bool']['must'].append(
+            _compile_exact_match_query("authors.last_name", parsed_name.last)
+        )
+        return nested_query
+
+    bool_must_query = list()
+    bool_must_query.append(_compile_exact_match_query("authors.last_name", parsed_name.last))
+    author_names = [re.split('\\.[A-Za-z]', name) for name in parsed_name.first_list]
+    author_names_retokenized = chain.from_iterable(author_names)
```

Comment:
I think this should be moved to `inspire-utils`. Having these hacks in two different places defeats the whole purpose of having that library in the first place. Now I'm thinking of it, why not move this whole mess (including tests!) to `inspire-utils`, and have a method like `ParsedName.generate_es_query()` instead of having it in two places?

Reply:
Yes, it would make sense, but which tests do you have in mind? Those, or also those from query parser?

Reply:
Those from the query parser. This does lots of different stuff, and it would be good to have the tests documenting the features.